src/_pytest/config/__init__.py

```diff
@@ -899,6 +900,20 @@ class InvocationParams:
         dir: Path
         """The directory from which :func:`pytest.main` was invoked."""
 
+    @final
+    class ArgsSource(enum.IntEnum):
+        """Indicates the source of the test arguments.
+
+        .. versionadded:: 7.2
+        """
+
+        #: Command line arguments.
+        ARGS = 0
+        #: Invocation directory.
+        INCOVATION_DIR = 1
+        #: 'testpaths' configuration value.
+        TESTPATHS = 2
```

Comment:
Could use

```suggestion
        #: Command line arguments.
        ARGS = enum.auto()
        #: Invocation directory.
        INCOVATION_DIR = enum.auto()
        #: 'testpaths' configuration value.
        TESTPATHS = enum.auto()
```

([Python docs](https://docs.python.org/3/library/enum.html#using-auto)), since we don't care about the value.